plugins/gimp/file-jxl-load.cc

```diff
@@ -81,9 +92,15 @@ bool LoadJpegXlImage(const gchar *const filename, gint32 *const image_id) {
         return false;
       }
 
+      xsize = info.xsize;
+      ysize = info.ysize;
+      if (info.have_animation) {
+        animation = info.animation;
+        tick_duration = animation.tps_numerator / animation.tps_denominator;
```

Comment:
Looks like this will need a cast to avoid integer division.

Reply:
It would be denominator / numerator, and the units are in seconds. I'm not sure the units on `tick_duration` but double check to see if it's in seconds or milliseconds or something.

Reply:
It's a bad variable name. I'm not using it as a duration.

Reply:
In that case, when creating the layer name, I think you need to divide by `tps_factor` rather than multiplying.  See attached for example.  Each frame should be 2000ms.
[rgb.jxl.zip](https://github.com/libjxl/libjxl/files/9944131/rgb.jxl.zip)

Reply:
Thanks, that's right.

Reply:
Fixed by [f150585](https://github.com/libjxl/libjxl/pull/1863/commits/f150585111c933a8396cb53013c4e47e1375baad).

![Screenshot from 2022-11-05 13-01-01](https://user-images.githubusercontent.com/838783/200132205-c9d83b0f-e6bc-4997-b56e-7063171fb5c0.png)